src/utils/autoconnect.ts

```diff
@@ -0,0 +1,70 @@
+import Web3Connect from 'web3connect'
+import { Provider, WalletConnectProvider as WCProvider } from '@gnosis.pm/dapp-ui'
+import WalletConnectProvider from '@walletconnect/web3-provider'
+import { delay } from 'utils'
+import { INFURA_ID, STORAGE_KEY_LAST_PROVIDER } from 'const'
+import { WalletApi } from 'types'
+
+const getWCIfConnected = async (): Promise<WCProvider | null> => {
+  const provider = new WalletConnectProvider({
+    infuraId: INFURA_ID,
+  })
+
+  // TODO: listen for specific websocket events when they are available
+  // hack to wait while WC tries connection with last session's key and uri
+  await delay(1000)
+
+  if (!provider.wc.connected) return null
+
+  try {
+    await provider.enable()
+  } catch (error) {
+    console.warn('Error reestablishing previous WC connection', error)
+    return null
+  }
+
+  return provider
+}
+
+export const getLastProvider = (): Promise<Provider | null> => {
+  const lastProviderName = localStorage.getItem(STORAGE_KEY_LAST_PROVIDER)
```

Comment:
is this synchronous? I thought localStorage had an async api

Reply:
or wait is that `localForage` :thinking:

Reply:
IndexDB is async, localStorage is sync